codegen/templates/templates.go

```diff
@@ -24,6 +24,7 @@ var CurrentImports *Imports
 
 type Options struct {
 	PackageName     string
+	Template        string
```

Comment:
This could use a comment describing the pretty large change in behaviour.

Reply:
@vektah I noticed a lot of the code is uncommented so I took the liberty to add documentation to what I know about the plugin system so far. Feel free to suggest edits/additions/removals.